### PR TITLE
Add a couple of auxiliary variable groups

### PR DESCRIPTION
**Pull request by mwarusz**
This PR adds three groups of auxiliary variables:
- `LayerThicknessAuxVars`
- `KineticAuxVars` (kinetic energy and divergence)
- `VorticityAuxVars`

Each group has tests comparing with analytical values. 

Checklist
* [x] Documentation:
  * [x] User's Guide has been updated
  * [x] Developer's Guide has been updated
  * [x] Documentation has been [built locally](https://e3sm-project.github.io/Omega/develop/devGuide/BuildDocs.html) and changes look as expected
* [ ] Testing
  * [x] CTest unit tests for new features have been added per the approved design.

### CODE REVIEW

**Comment by mwarusz** [comment]
This PR is ready for review. There are changes to the horizontal operators tests because I added a helper function for checking error measures.

File: components/omega/doc/userGuide/AuxiliaryVariables.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+(omega-user-aux-vars)=
+
+# Auxiliary Variables
+
+Omega needs to compute a couple of auxiliary variables at every model time step
+in order to advance the model state. Some auxiliary variables can be computed in
+different ways, and the user can specify that in the input configuration file.
+For example, the value of thickness used in advective flux can be centered or upwind.
+```yaml
+    advection:
```
Comment (philipwjones) [inline components/omega/doc/userGuide/AuxiliaryVariables.md:10]: I think these vars should be capitalized?

**Review by philipwjones — APPROVED** [review]
This looks good and passes unit testing for my usual Chrysalis (Intel) and Frontier (CPU/GPU) testing. Just had one minor doc change. Also, you will need to update unit tests based on the recent merge of Youngsung's change to unit testing.

**Comment by mwarusz** [comment]
@brian-oneill I have updated this PR to add a variable group for computing velocity del2.

**Review by brian-oneill — APPROVED** [review]
Everything looks great, thanks for all the work on this. Ran the test successfully on Perlmutter CPU & GPU as well.

**Review by hyungyukang — APPROVED** [review]
@mwarusz , many thanks for your efforts on this. All look good to me. I've been testing this PR by building a shallow water mode, where the kinetic energy, divergence, and vorticity-related variables (their interpolation to edges as well) are computed by this PR. As expected, the convergence rate with horizontal resolutions obtained by the nonlinear manufactured solution shows the second-order accuracy (shown in Figure).
image

One suggestion is to change the default option of `FluxThickEdgeChoice` from `Upwind`, which makes the model more dissipative and complicates obtaining accurate order slopes, to `Center`: https://github.com/E3SM-Project/Omega/pull/90/files#diff-9a33d13cc6fadae002857361b6673c9bbf836eddb5daa0222874b79262f17668R20
Of course, we can change the option based on tests, so it is not necessary to change it at this moment.

**Comment by mwarusz** [comment]
> One suggestion is to change the default option of FluxThickEdgeChoice from Upwind, which makes the model more dissipative and complicates obtaining accurate order slopes, to Center

@hyungyukang Done.

**Pull request merged**